test/unit/SchedulerSubsystem/SchedulerTest.java

```diff
@@ -0,0 +1,58 @@
+package unit.SchedulerSubsystem;
+
+import Networking.Direction;
+import Networking.Messages.DestinationEvent;
+import Networking.Messages.ElevatorStateEvent;
+import Networking.Receivers.DMA_Receiver;
+import Networking.Receivers.Receiver;
+import Networking.Transmitters.DMA_Transmitter;
+import Networking.Transmitters.Transmitter;
+import SchedulerSubsystem.Scheduler;
+import java.util.ArrayList;
+import org.junit.jupiter.api.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Unit tests for class Scheduler.
+ */
+public class SchedulerTest {
+
+    DMA_Receiver receiver, mockFloorReceiver, mockElevatorReceiver;
+    DMA_Transmitter mockTransmitterToFloor, mockTransmitterToElevator;
+    Scheduler s;
+    @BeforeEach
+    public void setUp() {
+        receiver = new DMA_Receiver();
+        mockFloorReceiver = new DMA_Receiver();
+        mockElevatorReceiver = new DMA_Receiver();
+        mockTransmitterToFloor = new DMA_Transmitter(mockFloorReceiver);
+        mockTransmitterToElevator = new DMA_Transmitter((mockElevatorReceiver));
+        s = new Scheduler(receiver, mockTransmitterToFloor, mockTransmitterToElevator);
+    }
+
+    /**
+     * Positive test for isElevatorStopping.
+     */
+    @Test
+    @DisplayName("isElevatorStopping positive test")
+    public void testPositiveIsElevatorStopping() {
```

Comment:
It's great to have a test for this method, but please take a look at this discussion board post:

https://brightspace.carleton.ca/d2l/le/224517/discussions/threads/530966/View

We should not be making private methods public purely to test them. If you would like, you can try to implement one of the methods mentioned in the discussion board to test that function.

Reply:
isElevatorStopping is OK as public. Considering the overall team's capacity, I don't agree that the added complexity in the given suggestions are worth the effort implementing. We can leave a comment  in the JavaDocs mentioning how if we had more time we could make it private!

I've refactored the other methods to stay private (eg. processMessage). I simply added a setter method for floorRequests to enable setting state in tests, as opposed to trying to set state by publishing events!